src/test/kotlin/org/jetbrains/gradle/ext/SerializationTests.kt

```diff
@@ -551,6 +551,93 @@ class SerializationTests {
 
   }
 
+  @Test fun `test multiple BeforeRunTasks of same type`() {
+    val beforeRun = JavaRunConfiguration.createBeforeRun(myProject)
+    val addCount = 2
+    listOf(Make::class, GradleTask::class, BuildArtifact::class).forEach { type ->
+      repeat(addCount) { i ->
+        val name = "${type.simpleName}$i"
+        beforeRun.create(name, type.java)
+        // Checks the name equals with the given $name.
+        assertEquals(name, beforeRun.findByName(name)?.getName())
```

Comment:
We still need to check this name even we made the json output tests, because it won't be serialized by `BeforeRunTask#toMap()`